lib/BC/SleighLifter.cpp

```diff
@@ -1101,6 +1091,20 @@ std::unordered_set<OpCode>
         CPUI_INT_SLESS,   CPUI_INT_LESSEQUAL, CPUI_INT_SLESSEQUAL,
         CPUI_INT_SBORROW, CPUI_INT_SCARRY,    CPUI_INT_CARRY};
 
+std::map<OpCode, SleighLifter::PcodeToLLVMEmitIntoBlock::BinaryOperator>
+    SleighLifter::PcodeToLLVMEmitIntoBlock::BOOL_BINARY_OPS = {
+        {OpCode::CPUI_BOOL_AND,
+         [](llvm::Value *lhs, llvm::Value *rhs, llvm::IRBuilder<> &bldr) {
```

Comment:
What is an upside of `std::map` (or `std::unordered_map`) compared to a function with a `switch`?

Reply:
This is fixing an ordering issue where in the bool function handling it would attempt to lift a value as a bool before determining if the bool ops apply. Instead now we check if we have a supported lifting operation, then lift the ops, then apply the function. This avoids having a value lifted as a boolean when it potentially should not be lifted as a boolean. An alternative is to have a list of BOOL_OPs that are checked to see if we need to do a bool lift, then lift the operand, then switch on the bool_ops again deciding what to do with that bool

Reply:
This is all pretty subtle and we should figure out a way to rework the lifter so bugs like this dont happen

Reply:
Well, add this as a comment to code then, so the next reader does not have to ponder as did I.

Reply:
Bump.